### PR TITLE
Wire up FirstChanceException

### PR DESCRIPTION
**Pull request by DnlHarvey**
The new targeting pack should allow us to call SetAppDomain which will enable us to stash an AppDomain that gets passed back to the FirstChanceException event

### CODE REVIEW

**Comment by DnlHarvey** [comment]
Wait I misread the diffs, the DataContractSerializer diffs should already be in.

File: src/System.Runtime.Extensions/src/System/AppDomain.cs [hunk 1]
```diff
@@ -58,7 +58,13 @@ public string FriendlyName
 
         public event EventHandler<FirstChanceExceptionEventArgs> FirstChanceException
         {
-            add { AppContext.FirstChanceException += value; }
+            add 
+            { 
+#if uapaot
+                AppContext.SetAppDomain(this);
```
Comment (krwq) [inline src/System.Runtime.Extensions/src/System/AppDomain.cs]: is this thread safe?

Comment (DnlHarvey) [inline src/System.Runtime.Extensions/src/System/AppDomain.cs]: Threads can race to set the appdomain, but there should only ever be one.

**Comment by DnlHarvey** [comment]
@dotnet-bot test OSX x64 Debug Build

**Pull request merged**